MyBookShelf/app/src/main/java/shuvalov/nikita/mybookshelf/MainActivity.java

```diff
@@ -0,0 +1,102 @@
+package shuvalov.nikita.mybookshelf;
+
+
+import android.graphics.Color;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.BaseAdapter;
+import android.widget.ImageView;
+import android.widget.ListView;
+import android.widget.TextView;
+
+
+import java.util.ArrayList;
+
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity {
+    BaseAdapter mBookAdapter;
+
+    List<Book> mBookList;
+    ListView listView;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        listView = (ListView) findViewById(R.id.listView);
+
+
+        //Use helper method to add books to the list
+        mBookList = generateBooks();
+        
+        mBookAdapter = new BaseAdapter() {
+            @Override
+            public int getCount() {
+                return mBookList.size();
+            }
+
+            @Override
+            public Object getItem(int position) {
+                return mBookList.get(position);
+            }
+
+            @Override
+            public long getItemId(int position) {
+                return position;
+            }
+
+            @Override
+            public View getView(int position, View convertView, ViewGroup parent) {
+                if(convertView == null){
+                    LayoutInflater layoutInflater = LayoutInflater.from(MainActivity.this);
+                    convertView = layoutInflater.inflate(R.layout.book_form, null);
+                }
+                TextView textView1 = (TextView) convertView.findViewById(R.id.title);
+                TextView textView2 = (TextView) convertView.findViewById(R.id.author);
+                TextView textView3 =(TextView) convertView.findViewById(R.id.date);
+                ImageView imgV = (ImageView) convertView.findViewById(R.id.bookImg);
+                textView1.setText(mBookList.get(position).getTitle());
+                textView2.setText(mBookList.get(position).getAuthor());
+                textView3.setText(mBookList.get(position).getDate());
+                imgV.setImageResource(mBookList.get(position).getImg());
```

Comment:
Good call saving just the resource ID as an int in the `Book` object. You'll probably want to do something similar when we learn about databases - rather than store the actual image in a database, you can just store the resource id (or url if it's remote).